python/meep.i

```diff
@@ -0,0 +1,99 @@
+/* Copyright (C) 2005-2017 Massachusetts Institute of Technology  
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software Foundation,
+ *  Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+ */
+
+%module(directors="1") meep
+
+%{
+#define SWIG_FILE_WITH_INIT
+#include "meep/vec.hpp"
+#include "meep.hpp"
+using namespace meep;
+%}
+
+%include "numpy.i"
+
+%{
+#if NPY_API_VERSION < 0x00000007
```

Comment:
Note that this is for [NumPy < 1.7](https://github.com/numpy/numpy/commit/5ede7c7bc2e9342269e708fd183c5311917b9169), i.e. numpy versions more than 5 years old; can probably omit this check nowadays?